File: src/java/nz/co/searchwellington/controllers/LoggedInUserFilter.java
package nz.co.searchwellington.controllers;

import javax.servlet.http.HttpServletRequest;

import nz.co.searchwellington.model.User;
import nz.co.searchwellington.repositories.UserRepository;

import org.apache.log4j.Logger;

public class LoggedInUserFilter {
	
	
	Logger log = Logger.getLogger(LoggedInUserFilter.class);
	   	
	private User loggedInUser;
	private UserRepository userDAO;
		
	public LoggedInUserFilter() {
	}

	public LoggedInUserFilter(UserRepository userDAO) {	
		this.loggedInUser = null;
		this.userDAO = userDAO; 
	}
	
	public void loadLoggedInUser(HttpServletRequest request) {		 
		if (request.getParameter("apikey") != null) {
			final String apiKey = request.getParameter("apikey");
			log.debug("Looking for an user by api key: " + apiKey);
			User user = userDAO.getUserByApiKey(apiKey);
			if (user != null) {
				this.loggedInUser = user;
				return;
			}
		}
		
		log.info("Looking for logged in user in session");
		if (request.getSession().getAttribute("user") != null) {
			User sessionUser =  (User) request.getSession().getAttribute("user");
			log.debug("Found user on session: " + sessionUser);
			this.loggedInUser = sessionUser;
			return;
		}
				
		this.loggedInUser = null;
	}

	public User getLoggedInUser() {
		return loggedInUser;
	}
	
}

File: src/main/scala/nz/co/searchwellington/controllers/LoggedInUserFilter.scala
package nz.co.searchwellington.controllers

import javax.servlet.http.HttpServletRequest
import nz.co.searchwellington.model.User
import org.apache.commons.logging.LogFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.{Scope, ScopedProxyMode}
import org.springframework.stereotype.Component

@Component("loggedInUserFilter")
@Scope(value = "request", proxyMode = ScopedProxyMode.TARGET_CLASS)
class LoggedInUserFilter @Autowired()(request: HttpServletRequest) {

  private val log = LogFactory.getLog(classOf[LoggedInUserFilter])

  def getLoggedInUser: Option[User] = getUserFromSession(request)

  def setLoggedInUser(request: HttpServletRequest, user: User): Unit = {
    log.info("Setting signed in user: " + user)
    request.getSession.setAttribute("user", user)
  }

  def clearLoggedInUser(request: HttpServletRequest): Unit = {
    request.getSession.setAttribute("user", null)
  }

  def isSignedIn: Boolean = getLoggedInUser.nonEmpty

  def getLoggedinUserProfileName: String = {
    getLoggedInUser.flatMap(_.profilename).orNull
  }

  def getLoggedInUserOrNull: User = {
    getLoggedInUser.orNull
  }

  private def getUserFromSession(request: HttpServletRequest): Option[User] = {
    val sessionUser = request.getSession.getAttribute("user").asInstanceOf[User]
    Option(sessionUser)
  }

}
